backend/src/schema/resolvers/login.spec.js

```diff
@@ -0,0 +1,80 @@
+const { userData } = require("../../data");
+
+const { gql } = require("apollo-server");
+const { createTestClient } = require("apollo-server-testing");
+
+const { constructTestServer } = require("../../utils/__utils");
+
+const POST_LOGIN = gql`
+  mutation login($username: String) {
+    login(username: $username) {
+      isLoggedIn
+      token
+    }
+  }
+`;
+
+const POST_SIGNUP = gql`
+  mutation signup($username: String) {
+    signup(username: $username) {
+      username
+      isLoggedIn
+      token
+    }
+  }
+`;
+
+let mutate;
+
+beforeAll(() => {
+  const { testServer } = constructTestServer();
+  mutate = createTestClient(testServer).mutate;
+});
+
+describe("Mutations", () => {
+  it("receiving token on login", async () => {
+    let res = await mutate({
+      mutation: POST_LOGIN,
+      variables: {
+        username: userData[0].username
+      }
+    });
+    expect(res.data.login.isLoggedIn).toBeTruthy();
+  });
+
+  it("error wrong username", async () => {
+    let res = await mutate({
+      mutation: POST_LOGIN,
+      variables: {
+        username: "Not A Human connected!"
+      }
+    });
+    expect(res).toMatchObject({
```

Comment:
You can combine the two statements and get much better expectation failure messages.

```suggestion
    await expect(await mutate({ /* ... */ })).resolves.toMatchObject({ /* ... */ })
```

Reply:
I just tried this and the expectation failure message was much worse and not really readable.

with our implementation:
```
expect(received).toMatchObject(expected)

    - Expected
    + Received

      Object {
    -   "data": null,
    -   "errors": Array [
    -     Object {
    -       "message": "There is no such user, you fool!",
    +   "data": Object {
    +     "login": Object {
    +       "isLoggedIn": true,
    +       "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6Ik1heCIsImlhdCI6MTU3NzQ0MzUwNX0.x52nfDRLz_WWfn5SbAYnnJILQTU_wzimTqwU2vsUzeA",
    +     },
        },
    -   ],
    +   "errors": undefined,
      }
```

vs. Your Suggestion:
```
 expect(received).resolves.toMatchObject()

    Matcher error: received value must be a promise

    Received has type:  object
    Received has value: {"data": {"login": {"isLoggedIn": true, "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6Ik1heCIsImlhdCI6MTU3NzQ0MzYzMH0._Uxsav0hJDwon34RDa13k3tKX_dSddayAPsY-i-kY_U"}}, "errors": undefined, "extensions": undefined, "http": {"headers": {Symbol(map): {}}}}

      61 |           username: "Max"
      62 |         }
    > 63 |       })).resolves.toMatchObject({
         |                    ^
      64 |         data: null,
      65 |         errors: [{ message: "There is no such user, you fool!" }]
      66 |       });
```

Reply:
Oh yes, I made a copy+paste error. Of course, the `await expect(promise).resolves...` must be a promise, I forgot to remove the await by accident:
```js
    await expect(mutate({ /* ... */ })).resolves.toMatchObject({ /* ... */ })
```

Reply:
Ah ok. it works now.
But it return the same expectation  failure message as ours. So I don't really the the benefit except having a variable less.